packages/toolpad-app/src/components/Home.tsx

```diff
@@ -112,45 +120,178 @@ function AppDeleteDialog({ app, onClose }: AppDeleteDialogProps) {
-        >
-          Edit
-        </Button>
-        <Button size="small" disabled={!app} onClick={onDelete}>
-          delete
-        </Button>
-      </CardActions>
-    </Card>
+    <React.Fragment>
+      <Card sx={{ gridColumn: 'span 1' }}>
+        <CardHeader
+          action={
+            <IconButton
+              aria-label="settings"
+              aria-controls={menuOpen ? 'basic-menu' : undefined}
+              aria-haspopup="true"
+              aria-expanded={menuOpen ? 'true' : undefined}
+              onClick={handleMenuClick}
+            >
+              <MoreVertIcon />
+            </IconButton>
+          }
+          disableTypography
+          title={
+            editingTitle ? (
+              <TextField
+                variant="standard"
+                size="small"
+                inputRef={appTitleInput}
+                sx={{ paddingBottom: '4px' }}
+                InputProps={{ sx: { fontSize: '1.5rem', height: '1.5em' } }}
+                onKeyUp={handleAppTitleInput}
+                onBlur={handleAppTitleBlur}
+                defaultValue={appTitle}
+              />
+            ) : (
+              <Typography gutterBottom variant="h5" component="div">
+                {app ? appTitle : <Skeleton />}
+              </Typography>
+            )
+          }
+          subheader={
+            <Typography variant="body2" color="text.secondary">
+              {app ? `Created: ${app.createdAt.toLocaleDateString('short')}` : <Skeleton />}
+            </Typography>
+          }
+        />
+
+        <CardActions>
+          <Button
+            size="small"
+            component="a"
+            href={app ? `/_toolpad/app/${app.id}/editor` : ''}
+            disabled={!app}
+          >
+            Edit
+          </Button>
+          <Button size="small" component="a" href={app ? `deploy/${app.id}` : ''} disabled={!app}>
+            View
+          </Button>
+        </CardActions>
+      </Card>
+      <Menu
+        id="basic-menu"
+        anchorEl={menuAnchorEl}
+        open={menuOpen}
+        onClose={handleMenuClose}
+        MenuListProps={{
+          'aria-labelledby': 'basic-button',
+          dense: true,
+        }}
+      >
+        <MenuItem onClick={handleRenameClick}>
+          <DriveFileRenameOutlineIcon sx={AppCardMenuIconStyles} />
+          Rename
+        </MenuItem>
+        <MenuItem onClick={handleMenuClose}>
+          <FileCopyIcon sx={AppCardMenuIconStyles} /> Duplicate
+        </MenuItem>
+        <MenuItem onClick={handleDeleteClick}>
+          <DeleteIcon sx={AppCardMenuIconStyles} />
+          Delete
+        </MenuItem>
+      </Menu>
+    </React.Fragment>
   );
 }
 
 export default function Home() {
-  const { data: apps = [], status, error } = client.useQuery('getApps', []);
+  const [apps, setApps] = React.useState<App[]>([]);
+
+  const { status, error } = client.useQuery('getApps', [], {
+    onSuccess: (data: App[]) => setApps(data),
```

Comment:
not necessary, use `refetch` to refresh the apps

Reply:
The issue with performing a refetch was that it modified the order of the apps returned, so every rename operation caused the cards to switch around

Reply:
We shouldn't order them alphabetically then, we should order by "last edited".

Reply:
By "last edited" I mean the app dom, not the database record for the app. We'll need to intoruduce another field in the db schema to express this.

Reply:
Should I use `refetch` for the scope of the current PR and add this in a separate PR?

Reply:
Yep, that's good 👍

Reply:
not sure how we're doing it, but ideally with GraphQL caching if we return `id` and `name` from the mutation it would just update the name of the app in the cache, and we wouldn't need to refetch?

Reply:
`refetch` in the context of `react-query` is essentially just invalidating its cache for that list query. You don't need to do anything else in terms of updating UI.